website/docs/expansion.md

```diff
@@ -0,0 +1,298 @@
+resource (the Deployment in this example).
+
+To use this functionality, first specify which resources should be "expanded"
+into mock resource(s) by creating an `ExpansionTemplate` custom resource. This
+specifies the GVKs of the workload resources, the GVK of
+the resultant mock resource, as well as which subfield of the workload resource
+should be used to expand the mock resource (i.e. `spec.template` for most
+Pod-generating resources).
+
+In some cases, it may not be possible to build an accurate representation of a
+mock resource by looking at the workload resource alone. For example, suppose a
+cluster is using Istio, which will inject a sidecar container on specific
+resources. This sidecar configuration is not specified in the config of the
+workload resource (i.e. Deployment), but rather injected by Istio's webhook. In
+order to accurately represent mock resources modified by controllers or
+webhooks, Gatekeeper leverages its
+[Mutations](https://open-policy-agent.github.io/gatekeeper/website/docs/mutation)
+feature to allow mock resources to be manipulated into their desired form. In
+the Istio example, `Assign` and `ModifySet` mutators could be configured to
+mimic Istio sidecar injection. For further details on mutating mock resources
+see the [Math Source](#match-source) section below, or to see a working example,
+see the [Example](#example) section.
+
+Any resources configured for expansion will be expanded by both the validating
+webhook and
+[Audit](https://open-policy-agent.github.io/gatekeeper/website/docs/audit). This
+feature will only be enabled if a user creates a `ExpansionTemplate` that
+targets some resource that exists on the cluster.
+
+Note that the accuracy of enforcement depends on how well the mock resource
+resembles the real thing. Mutations can help with this, but 100% accuracy is
+impossible because not all fields can be predicted. For instance, deployments
+create pods with random names. Inaccurate mocks may lead to over or under
+enforcement. In the case of under enforcement, the resultant pod should still be
+rejected. Finally, non-state-based policies (those that rely on transient
+metadata such as requesting user or time of creation) cannot be enforced
+accurately. This is because such metadata would necessarily be different when
+creating the resultant resource. For example, a deployment is created using the
+requesting user's account, but the pod creation request comes from the service
+account of the deployment controller.
+
+If, for any reason, you want to exempt expanded resources from a specific
+constraint, look at the [Match Source](#match-source) section below.
+
+## Configuring Expansion
+
+Expansion behavior is configured through the `ExpansionTemplate` CR. Optionally,
+users can create `Mutation` CRs to customize how resources are expanded.
+Mutators with the `source: Generated` field will only be applied when expanding
+workload resources, and will not mutate real resources on the cluster.
```

Comment:
Might be good to mention what is the default `source` if not provided and how it applies to both generated and original.